recipes/functions-framework-cpp/all/conanfile.py

```diff
@@ -22,64 +25,59 @@ class FunctionsFrameworkCppConan(ConanFile):
         "shared": False,
         "fPIC": True,
     }
-
-    generators = "cmake", "cmake_find_package_multi", "cmake_find_package"
     short_paths = True
-    _cmake = None
 
     @property
-    def _source_subfolder(self):
-        return "source_subfolder"
+    def _min_cppstd(self):
+        return 17
 
     @property
-    def _is_msvc(self):
-        return str(self.settings.compiler) in ["Visual Studio", "msvc"]
+    def _compilers_minimum_version(self):
+        return {
+            "gcc": "9",
+            "Visual Studio": "15.7",
+            "msvc": "190",
+            "clang": "7",
+            "apple-clang": "11",
+        }
 
     def export_sources(self):
-        self.copy("CMakeLists.txt")
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            self.copy(patch["patch_file"])
+        export_conandata_patches(self)
 
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
+            self.options.rm_safe("fPIC")
 
-    def requirements(self):
-        self.requires("abseil/20211102.0")
-        self.requires("boost/1.78.0")
-        self.requires("nlohmann_json/3.10.5")
+    def layout(self):
+        cmake_layout(self, src_folder="src")
 
-    @property
-    def _compilers_minimum_version(self):
-        return {
-            "gcc": "9",
-            "Visual Studio": "15.7",
-            "clang": "7",
-            "apple-clang": "11",
-        }
+    def requirements(self):
+        self.requires("abseil/20230125.3")
+        self.requires("boost/1.82.0")
+        self.requires("nlohmann_json/3.11.2")
```

Comment:
Seems like a missing `transitive_headers=True` trait?
```suggestion
        self.requires("nlohmann_json/3.11.2", transitive_headers=True)
```